common/lib/dependabot/clients/azure.rb

```diff
@@ -48,7 +48,11 @@ def fetch_commit(_repo, branch)
           "/_apis/git/repositories/" + source.unscoped_repo +
           "/stats/branches?name=" + branch)
 
-        JSON.parse(response.body).fetch("commit").fetch("commitId")
+        commit = JSON.parse(response.body).fetch("commit", nil)
+
+        raise NotFound unless commit
```

Comment:
Could we instead check if the `response.status` is 400 instead of checking if the commit is present?

Reply:
Updated to raise error when response.status == 400